projects/sandbox/data_generation/data_generation/bilby_injections.py

```diff
@@ -0,0 +1,177 @@
+    # duration which will enforce that the signal
+    # lies in the center of the kernel
+    np.savetxt(
+        datadir / "start_times.txt", signal_times - (bilby_duration / 2)
+    )
+    min_duration = (
+        signal_times[-1] - signal_times[0] + waveform_duration // 2 + buffer
+    )
+
+    segment_start, segment_stop = query_segments(
+        segment_names,
+        start,
+        stop,
+        min_duration,
+    )[0]
+
+    background_dict = download_data(
+        ifos, frame_type, channel, sample_rate, segment_start, segment_stop
+    )
+    df = 1 / waveform_duration
+    if gaussian:
+        logging.info(
+            "Generating gaussian noise from psd for injection background"
+        )
+        for ifo in ifos:
+            duration = len(background_dict[ifo]) / sample_rate
+            psd = normalize_psd(background_dict[ifo], df, sample_rate)
+            data = noise_from_psd(psd, df, duration, sample_rate)
+            background_dict[ifo] = TimeSeries(data, dt=1 / sample_rate)
+
+    # instantiate prior, sample, and generate signals
+    prior = prior()
+    parameters = prior.sample(n_samples)
+    signals = generate_gw(
+        parameters,
+        sample_rate,
+        waveform_duration,
+        waveform,
+        waveform_arguments=waveform_arguments,
+        parameter_conversion=parameter_conversion,
+    )
+
+    plus, cross = signals.transpose(1, 0, 2)
+    plus = torch.Tensor(plus)
+    cross = torch.Tensor(cross)
+
+    # project raw polarizations onto interferometers
+    # with sampled sky localizations
+    tensors, vertices = get_ifo_geometry(*ifos)
+
+    # dec is declination
+    # psi is polarization angle
+    # phi is relative azimuthal angle between source and earth
+    dec = torch.Tensor(parameters["dec"])
+    psi = torch.Tensor(parameters["psi"])
+    phi = torch.Tensor(parameters["ra"]) - np.pi
+
+    waveforms = compute_observed_strain(
+        dec,
+        psi,
+        phi,
+        tensors,
+        vertices,
+        sample_rate,
+        plus=plus,
+        cross=cross,
+    )
+
+    parameters["geocent_time"] = signal_times
+
+    waveforms = waveforms.numpy()
+    data_dict = TimeSeriesDict()
+    for i, (ifo, data) in enumerate(background_dict.items()):
+        # set start time of data to 0 for simplicity
+        data.t0 = 0
+        times = data.times.value
+
+        # inject waveforms into background and specified times
+        data = inject_into_background(
+            (times, data),
+            waveforms[:, i, :],
+            signal_times,
+        )
+
+        # package into gwpy timeseries and save as hdf5 files
+        data_dict[ifo] = TimeSeries(
+            data, dt=1 / sample_rate, channel=f"{ifo}:{channel}", t0=0
+        )
+
+    data_dict.write(
+        datadir / "bilby_timeseries.hdf5",
+        format="hdf5",
+        overwrite=True,
+    )
+    # save parameters as hdf5 file
+    with h5py.File(datadir / "bilby_injection_parameters.hdf5", "w") as f:
```

Comment:
Are we saving the data in a format that our sample project also understands?

Reply:
Yes